avm/server/src/runner.rs

```diff
@@ -76,7 +77,13 @@ impl AVMRunner {
         ttl: u32,
         current_peer_id: impl Into<String>,
         call_results: CallResults,
+        keypair: &KeyPair,
     ) -> RunnerResult<RawAVMOutcome> {
+        let key_format = keypair.key_format();
+        let key_bytes: Vec<u8> = keypair
+            .secret()
+            .unwrap_or_else(|_| panic!("failed to serialize key of format {:?}", key_format));
```

Comment:
why panic if you can return `Err`? I wouldn't expect `AVM::call` to ever panic

Reply:
It panics only if RSA key is provided. OTOH you are right, it is implementation detail.

We might remove RSA key support from `fluence-keypair`, making this method infallible.